gluten-core/src/main/scala/io/glutenproject/expression/UnaryOperatorTransformer.scala

```diff
@@ -206,7 +206,9 @@ class LengthTransformer(child: Expression, original: Expression)
 
   override def doTransform(args: java.lang.Object): ExpressionNode = {
     val childNode = child.asInstanceOf[ExpressionTransformer].doTransform(args)
-    if (!childNode.isInstanceOf[ExpressionNode]) {
+    if (!childNode.isInstanceOf[ExpressionNode]
+      || child.dataType.isInstanceOf[BinaryType]) {
+      // length(BinaryType) got incorrect result, so fall back to Vanilla Spark.
```

Comment:
Does vanilla Spark support binary type for this expression? The design is putting the backend-specific validation logic to the backend validator, instead of in Gluten.

Reply:
Yes, I've put this in the backend-velox  validation.